sys-utils/libcmt/include/libcmt/rollup.h

```diff
@@ -0,0 +1,161 @@
+/* Copyright Cartesi and individual authors (see AUTHORS)
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file
+ * @defgroup libcmt_rollup rollup
+ * Rollup abstraction layer
+ *
+ * Takes care of @ref libcmt_io_driver interactions, @ref libcmt_abi
+ * encoding/decoding and @ref libcmt_merkle tree interactions.
+ *
+ * Mocked version has support for simulating I/O via environment variables:
+ * @p CMT_INPUTS="0:input.bin,..." and verbose ouput with @p CMT_DEBUG=yes.
+ *
+ * Lets look at some code:
+ *
+ * @include doc/examples/rollup.c
+ *
+ * @ingroup libcmt
+ * @{ */
+#ifndef CMT_ROLLUP_H
+#define CMT_ROLLUP_H
+#include "abi.h"
+#include "buf.h"
+#include "io.h"
+#include "merkle.h"
+
+typedef struct cmt_rollup {
+    union cmt_io_driver io[1];
+    cmt_buf_t tx[1], rx[1];
```

Comment:
Why is tx and tx visible here? They also exist in `cmt_io_driver` and can be obtained by `cmt_io_get_tx` and `cmt_io_get_rx`. There is no API function in `rollup.h` that needs them, right?

Reply:
I'll remove it from here. This surfaced after simplifications.